src/implementations/vanilla/components/checkbox/_checkbox.scss

```diff
@@ -130,52 +118,70 @@ $_label-color: null !default;
 // Invalid
 .ecl-checkbox--invalid {
   @at-root .ecl-form-group & ~ .ecl-feedback-message {
-    margin-top: map.get(theme.$spacing, 'xs');
+    margin-top: map.get($theme, 'spacing', '2xs');
   }
 
   .ecl-checkbox__box {
-    border-color: $_invalid-border-color;
+    border-color: map.get($checkbox, 'invalid-border-color');
   }
 
   .ecl-checkbox__invalid {
-    font: map.get(theme.$font, 's');
-    margin-top: map.get(theme.$spacing, 'xs');
+    font: map.get($theme, 'font', 's');
+    margin-top: map.get($theme, 'spacing', 'xs');
   }
 }
 
 .ecl-checkbox--invalid
   .ecl-checkbox__label:hover
   .ecl-checkbox__box:not(.ecl-checkbox__box--disabled) {
-  border-color: $_invalid-border-color-hover;
+  border-color: map.get($checkbox, 'invalid-border-color-hover');
 }
 
 /* stylelint-disable-next-line plugin/selector-bem-pattern, selector-max-compound-selectors */
 .ecl-checkbox--invalid
   input:checked
   ~ .ecl-checkbox__label:hover
   .ecl-checkbox__box:not(.ecl-checkbox__box--disabled) {
-  background: map.get(theme.$color, 'red-130');
+  background: var(--ecl-color-error);
 }
 
 .ecl-checkbox__label:hover
   .ecl-checkbox__box:not(.ecl-checkbox__box--disabled) {
-  border-color: $_box-border-color-hover;
+  border-color: map.get($checkbox, 'box-border-color-hover');
+
+  @if map-has-key($checkbox, 'hover-circle') {
+    &::before {
+      @each $property, $value in map.get($checkbox, 'hover-circle') {
+        #{$property}: $value;
+      }
+    }
+  }
 }
 
 // Hover and default focus state
 .ecl-checkbox__input:focus-visible {
   ~ .ecl-checkbox__label {
     .ecl-checkbox__box:not(.ecl-checkbox__box--disabled) {
-      outline: 2px solid map.get(theme.$color, 'blue-100');
-      outline-offset: 2px;
+      @if map-has-key($checkbox, 'hover-circle') {
+        &::before {
+          @each $property, $value in map.get($checkbox, 'hover-circle') {
+            #{$property}: $value;
+          }
+        }
+      } @else {
+        outline: 2px solid map.get($theme, 'color' 'blue-100');
```

Comment:
We're not supposed to use color blue-100 anymore (or any non semantic colors). I assume that it should be --ecl-color-primary